src/services/InvoiceService.ts

```diff
@@ -61,23 +61,27 @@ export default class InvoiceService {
   }
 
   async getAllInvoices(params: ListParams): Promise<InvoiceListResponse> {
-    const findOptions: FindManyOptions<Invoice> = {
-      order: {
-        [params.sorting?.column ?? 'id']:
-        params.sorting?.direction ?? 'ASC',
-      },
-    };
+    const queryBuilder = this.repo.createQueryBuilder('invoice');
+
+    queryBuilder
+      .skip(params.skip)
+      .take(params.take)
+      .orderBy(`${queryBuilder.alias}.${params.sorting?.column ?? 'id'}`, params.sorting?.direction ?? 'ASC')
+      // initial where() to allow chaining andWhere() function calls
+      .where('1 = 1')
+    ;
+
+    if (params.search) {
+      addQueryBuilderSearch(queryBuilder, params.search, ['title', 'company.name']);
+    }
 
-    findOptions.where = addQueryWhereClause(params, ['title', 'company.name']);
+    if (params.filters && params.filters.length > 0) {
+      addQueryBuilderFilters(queryBuilder, params.filters);
+    }
 
     return {
-      list: await this.repo.find({
-        ...findOptions,
-        skip: params.skip,
-        take: params.take,
-      }),
-      count: await this.repo.count(findOptions),
-      lastSeen: await this.getTreasurerLastSeen(),
+      list: await queryBuilder.getMany(),
+      count: await queryBuilder.clone().getCount(),
```

Comment:
This will not work. On line 68 you set a "take" in the query, which limits the amount of query results. `getCount()` returns the number of matches, which will therefore be at most the value of `take`. You can fix this by applying the pagination explicitly on line 83 (so `await queryBuilder.skip(params.skip).take(params.take).getMany(),`). If I remember correctly, TypeORM returns a new builder object when you make changes to it, so should not need to manually clone.